### ISSUE TITLE
Duplicate sizes causes missing .is-size-* and .title.is-* declarations

### ISSUE THREAD

**Issue by kofigumbs** [jgthms/bulma#2802]
- This is about the Bulma **CSS framework**
- I'm using Bulma's **master branch**
- I am sure this issue is **not a duplicate**

### Description

Due to the way `@each` and `index()` are used in the size helpers, Bulma will only generate CSS for the first of any duplicate sizes definitions. A duplicate size definition would look like this:

```sass
$size-1: 100px
$size-2: 100px
@import "bulma"
```

This happens because `index()` always grabs the index of the _first_ matching value. Because this is a nuance specific to `index()`, this bug only affects 3 places in the Bulma codebase:

1. https://github.com/jgthms/bulma/blob/0c43b9e4663ae040d47d2a373ded5ba914b7d10f/sass/base/helpers.sass#L25-L28
1. https://github.com/jgthms/bulma/blob/0c43b9e4663ae040d47d2a373ded5ba914b7d10f/sass/elements/title.sass#L49-L52
1. https://github.com/jgthms/bulma/blob/0c43b9e4663ae040d47d2a373ded5ba914b7d10f/sass/elements/title.sass#L67-L70
### Steps to Reproduce

I setup an example repository with the compiled CSS where you can see the current behavior:

- Code: https://github.com/hkgumbs/bulma-size-bug-demo
- Demo: https://hkgumbs.github.io/bulma-size-bug-demo/
- TL;DR
  1. Set variables `size-1` and `size-2` to the same value (i.e. `100px`)
  1. Compile Bulma via the normal means
  1. Compiled CSS **does not** include declarations for `is-size-2`/`is-2`

### Expected behavior

Bulma would produce declarations for `is-size-1`/`is-1` _and_ `is-size-2`/`is-2`. All of these declarations would be set to the same value (`100px` in this case).

### Actual behavior

Bulma produces 2 declarations for `is-size-1`/`is-1` and 0 for `is-size-2`/`is-2`: https://github.com/hkgumbs/bulma-size-bug-demo/blob/master/output.css#L378-L382

**Comment by hkgumbs** [comment]
This is still an issue in `master`.
